In [30]:
import numpy as np
import pandas as pd
from scipy import stats
import seaborn as sns
import math 
import random
import matplotlib.pyplot as plt
from datetime import datetime
%matplotlib inline

In [31]:
# read in zillow property values for NYC
home_df = pd.read_csv('homes.csv')

In [32]:
len(home_df)

172

In [33]:
# drop unneeded columns
home_df.drop(home_df.columns[4:], axis=1, inplace=True)

In [34]:
# drop unneeded columns
home_df.drop(home_df.columns[1:3], axis=1, inplace=True)

In [35]:
# renamed columns to zip and values
home_df.rename(columns={'New York, NY - All Homes': 'Zip Code', 'Unnamed: 3': 'Value'}, inplace=True)

In [36]:
# test to make sure columns populated appropriately 
home_df.columns

Index(['Zip Code', 'Value'], dtype='object')

In [37]:
home_df.head(5)

Zip Code                    Value
0    Statistic  Zillow Home Value Index
1  Region Name                  Current
2     New York                 $676,000
3        10001               $1,892,700
4        10002                 $909,900

In [38]:
# delete the first three rows so that we only have zips and values
home_df.drop(home_df.index[:3], inplace=True)


In [39]:
# delete all rows with no value
home_df = home_df[home_df.Value != '---']

In [40]:
# test to see what the beginning of the dataset looks like
home_df.head(5)

Zip Code       Value
3    10001  $1,892,700
4    10002    $909,900
5    10003  $1,544,000
6    10004  $1,036,200
7    10005  $1,099,900

In [45]:
# parse out $ and convert currency str to float
home_df[home_df.columns[1:]] = home_df[home_df.columns[1:]].replace('[\$,]', '', regex=True,).astype(float)

In [47]:
#sort dataframe for highest values by zip code
home_df.sort_values(by='Value', ascending = False)


Zip Code      Value
14     10013  2895400.0
9      10007  2545200.0
13     10012  2359700.0
3      10001  1892700.0
15     10014  1737700.0
42     10069  1701300.0
11     10010  1630000.0
12     10011  1585500.0
5      10003  1544000.0
41     10065  1528200.0
20     10021  1489800.0
27     10028  1392100.0
107    11225  1371800.0
22     10023  1355000.0
21     10022  1245100.0
97     11215  1219300.0
99     11217  1206000.0
44     10128  1161000.0
19     10019  1158600.0
112    11231  1149500.0
111    11230  1120800.0
7      10005  1099900.0
43     10075  1098500.0
77     11101  1093800.0
100    11218  1087800.0
10     10009  1056200.0
120    11249  1047200.0
35     10036  1037900.0
6      10004  1036200.0
91     11209  1033200.0
..       ...        ...
134    11368   527700.0
64     10461   527700.0
151    11417   527200.0
165    11435   526600.0
68     10465   521100.0
153    11419   517000.0
54     10308   517000.0
58     10314   513100.0
154    11420   499400.0
56     10310   497300.0
117    11236   490800.0
72     10469   486200.0
73     10470   477700.0
161    11429   475300.0
146    11412   474300.0
156    11422   473200.0
147    11413   469900.0
90     11208   467100.0
94     11212   457500.0
164    11434   457400.0
145    11411   453400.0
166    11436   447300.0
163    11433   446400.0
69     10466   438400.0
75     10473   435000.0
70     10467   431300.0
48     10302   409800.0
168    11692   370700.0
49     10303   357300.0
65     10462   162300.0

[144 rows x 2 columns]

### We can see here that the 10 highest valued zip codes are the following: 10013, 10007, 10012, 10001, 10014, 10069, 10010, 10011, 10003, 10065  	